components/ai_chat/resources/untrusted_conversation_frame/components/assistant_response/tool_event.tsx

```diff
@@ -0,0 +1,135 @@
+import styles from './tool_event.module.scss'
+
+interface Props {
+  toolUseEvent: Mojom.ToolUseEvent
+  isEntryActive: boolean
+}
+
+// Content customizable for each known tool type
+interface ToolUseContent {
+  toolText: JSX.Element
+  tooltipContent: JSX.Element | null
+  statusIcon: JSX.Element
+  progressIcon: JSX.Element
+}
+
+function useUserChoiceToolContent(props: Props, toolInput: any): Partial<ToolUseContent> {
+  const context = useUntrustedConversationContext()
+  const content: Partial<ToolUseContent> = {}
+
+  if (props.toolUseEvent.output) {
+    // Tool already completed, don't allow further response
+    content.toolText = (
+      <ConversationAreaButton
+        onClick={() => {}}
+        isDisabled={true}
+        icon={<Icon className={styles.completedChoiceIcon} name='checkbox-checked' />}
+      >
+        {props.toolUseEvent.output[0]?.textContentBlock?.text}
+      </ConversationAreaButton>
+    )
+  } else {
+    content.progressIcon = <Icon name='help-outline' />
+
+    const handleChoice = (choice: string) => {
+      if (!props.isEntryActive) {
+        return
+      }
+      context.conversationHandler?.respondToToolUseRequest(
+        props.toolUseEvent.id, [createTextContentBlock(choice)]
+      )
+    }
+
+    content.toolText = (
+      <>
+        {toolInput?.choices?.map((choice: string, i: number) => (
+          <div key={i} className={styles.toolChoice}>
+            <ConversationAreaButton
+              className={classnames(styles.choice)}
+              isDisabled={!props.isEntryActive}
+              onClick={() => handleChoice(choice)}
+              icon={<div className={classnames(styles.choiceNumber)}>{i+1}</div>}
+            >
+              {choice}
+            </ConversationAreaButton>
+          </div>
+        ))}
+      </>
+    )
+  }
+
+  return content
+}
+
+export function useToolEventContent(props: Props): ToolUseContent {
+  const {toolUseEvent} = props
+
+  let content: ToolUseContent = {
+    toolText: <>{toolUseEvent.toolName}</>,
+    tooltipContent: null,
+    statusIcon: <Icon name="check-circle-outline" />,
+    progressIcon: <ProgressRing />
+  }
+
+  const input = React.useMemo(() => {
+    if (!toolUseEvent?.argumentsJson) {
+      return null
+    }
+    try {
+      return JSON.parse(toolUseEvent.argumentsJson)
+    } catch (e) {
+      return null
+    }
+  }, [toolUseEvent?.argumentsJson])
+
+  if (toolUseEvent.toolName === Mojom.USER_CHOICE_TOOL_NAME) {
+    content = { ...content, ...useUserChoiceToolContent(props, input)}
+  }
+
+  return content
+}
+
+
+export default function ToolEvent(props: Props) {
+  const {
+    progressIcon,
+    statusIcon,
+    toolText,
+    tooltipContent
+  } = useToolEventContent(props)
+
+  const isComplete = !!props.toolUseEvent.output
+  return (
+    <div className={classnames(styles.toolUse, isComplete && styles.toolUseComplete, `tool-${props.toolUseEvent.toolName}`)}>
```

Comment:
Do we need the `'tool-${props.toolUseEvent.toolName}'` class? We don't seem to use it for styling anywhere.

Reply:
I guess not for this tool

Reply:
Just a little confused about when we would use it - wouldn't the place to do the styling be in the per tool component?

Reply:
well actually it _is_ being used for the tool in this PR, see `tool_event.module.scss`:

```scss
...
  &:not(&.toolUseComplete):global(.tool-user_choice_tool) {
    .toolUseIcon {
      --leo-icon-color: var(--leo-color-primary-5);
      background-color: var(--leo-color-primary-40);
      border-color: var(--leo-color-primary-50);
    }
  }
...
```

I just thought it might be nice not to have the different tools also come with their own css rules. But who knows how this UI will end up being styled after we get a couple of tools.